src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/SecurityServiceDispatcher.cs

```diff
@@ -461,7 +461,7 @@ public override async Task DispatchAsync(RequestContext context)
             await serviceChannelDispatcher.DispatchAsync(securedMessage);
         }
 
-        public override Task DispatchAsync(Message message)
+        public override async Task DispatchAsync(Message message)
         {
```

Comment:
You can either still throw, or use Task.FromException. This is the slow way to achieve this.

Reply:
done